### PR TITLE
UIImage extensions for render mode 🗿

### PR DESCRIPTION
**Pull request by portellaa**
added to simple vars, one for original render mode and another one for
template

i used this in a project and i think they are useful, when we really need them.

### CODE REVIEW

File: Tests/AlicerceTests/UIKit/UIImageTestCase.swift [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        let mrMinder = imageFromFile(withBundleClass: UIImageTestCase.self,
+                                     name: "mr-minder",
+                                     type: "png")
+
+        let mrMinderOriginal = mrMinder.original
+
+        XCTAssertEqual(mrMinderOriginal.renderingMode, UIImageRenderingMode.alwaysOriginal)
+    }
+
+    func testOriginal_WhenHasImage_ItShouldReturnTheOriginalImage() {
```
Comment (jakewelton) [inline Tests/AlicerceTests/UIKit/UIImageTestCase.swift]: Ermm...... should these test names be the same? 😛

File: Tests/AlicerceTests/UIKit/UIImageTestCase.swift [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+final class UIImageTestCase: XCTestCase {
+
+    func testOriginal_WhenHasImage_ItShouldReturnTheOriginalImage() {
+        let mrMinder = imageFromFile(withBundleClass: UIImageTestCase.self,
+                                     name: "mr-minder",
+                                     type: "png")
+
+        let mrMinderOriginal = mrMinder.original
+
+        XCTAssertEqual(mrMinderOriginal.renderingMode, UIImageRenderingMode.alwaysOriginal)
```
Comment (jakewelton) [inline Tests/AlicerceTests/UIKit/UIImageTestCase.swift]: Not sure if you added this for clarity, but I don't think you need to specify `UIImageRenderingMode.alwaysOriginal`. You could just use the type inference, so `.alwaysOriginal`

Comment (portellaa) [inline Tests/AlicerceTests/UIKit/UIImageTestCase.swift]: i thought on that. honestly, i don't know what to say in this case, i just got the enum and that's it.
i will remove it 😉

File: Tests/AlicerceTests/UIKit/UIImageTestCase.swift (hunk 1 above)
Comment (portellaa) [inline Tests/AlicerceTests/UIKit/UIImageTestCase.swift]: yeap 😂

File: Tests/AlicerceTests/UIKit/UIImageTestCase.swift [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+        XCTAssertEqual(mrMinderOriginal.renderingMode, .alwaysOriginal)
+    }
+
+    func testOriginal_WhenHasImage_ItShouldReturnTheTemplateImage() {
+        let mrMinder = imageFromFile(withBundleClass: UIImageTestCase.self,
+                                     name: "mr-minder",
+                                     type: "png")
+
+        let mrMinderOriginal = mrMinder.template
```
Comment (jakewelton) [inline Tests/AlicerceTests/UIKit/UIImageTestCase.swift]: I'm guessing this name came along after the copy & paste. I'm guessing it should be `mrMinderTemplate`? :smile:

**Comment by p4checo** [comment]
Hi @portellaa, are you going to update this PR? 😇

**Comment by portellaa** [comment]
Did i miss something? the tests?

File: Tests/AlicerceTests/UIKit/UIImageTestCase.swift (hunk 3 above)
Comment (p4checo) [reply Tests/AlicerceTests/UIKit/UIImageTestCase.swift]: @portellaa the only thing I see that might be missing is this ☝️

Comment (portellaa) [reply Tests/AlicerceTests/UIKit/UIImageTestCase.swift]: ups 😂

Comment (portellaa) [reply Tests/AlicerceTests/UIKit/UIImageTestCase.swift]: you guys must be crazy, don't know what you are talking about 😝 

btw, fixed 😂

File: Tests/AlicerceTests/UIKit/UIImageTestCase.swift [hunk 4]
```diff
@@ -20,16 +20,16 @@ final class UIImageTestCase: XCTestCase {
 
         let mrMinderOriginal = mrMinder.original
 
-        XCTAssertEqual(mrMinderOriginal.renderingMode, UIImageRenderingMode.alwaysOriginal)
+        XCTAssertEqual(mrMinderOriginal.renderingMode, .alwaysOriginal)
     }
 
-    func testOriginal_WhenHasImage_ItShouldReturnTheOriginalImage() {
+    func testOriginal_WhenHasImage_ItShouldReturnTheTemplateImage() {
```
Comment (p4checo) [inline Tests/AlicerceTests/UIKit/UIImageTestCase.swift]: `testTemplate_...`?

Comment (portellaa) [reply Tests/AlicerceTests/UIKit/UIImageTestCase.swift]: 🤐

**Comment by p4checo** [comment]
go go go 🚀

**Pull request merged**